tree/ntuple/v7/src/RFieldMeta.cxx

```diff
@@ -835,7 +835,7 @@ void ROOT::Experimental::RField<TObject>::ReadGlobalImpl(NTupleSize_t globalInde
 
 void ROOT::Experimental::RField<TObject>::AfterConnectPageSource()
 {
-   if (GetTypeVersion() != 1) {
+   if (GetOnDiskTypeVersion() != 1) {
```

Comment:
(note for future reviewers: the in-memory class version is `assert`ed in the constructor)